core/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalMaxTests.java

```diff
@@ -19,20 +19,30 @@
 
 package org.elasticsearch.search.aggregations.metrics;
 
+import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.rest.action.search.RestSearchAction;
 import org.elasticsearch.search.DocValueFormat;
 import org.elasticsearch.search.aggregations.InternalAggregationTestCase;
 import org.elasticsearch.search.aggregations.metrics.max.InternalMax;
 import org.elasticsearch.search.aggregations.pipeline.PipelineAggregator;
 
+import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+
 public class InternalMaxTests extends InternalAggregationTestCase<InternalMax> {
     @Override
     protected InternalMax createTestInstance(String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
-        return new InternalMax(name, randomDouble(),
-                randomFrom(DocValueFormat.BOOLEAN, DocValueFormat.GEOHASH, DocValueFormat.IP, DocValueFormat.RAW), pipelineAggregators,
+        return new InternalMax(name, frequently() ? randomDouble() : Double.NEGATIVE_INFINITY,
```

Comment:
I can add that, the problem is that we cannot distinguish between NEGATIVE_INFINITY and POSITIVE_INFINITY once we parse from REST because both values are renderes out as `null`. We will need special cases in the test assertions for that.